### PR TITLE
Raise error if grants deadline is closed

### PR DESCRIPTION
**Pull request by estyxx**

### CODE REVIEW

File: backend/api/grants/tests/test_send_grant.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    variables = {**defaults, **kwargs}
+
+    response = client.query(document, variables=variables)
+
+    return response
+
+
+@pytest.mark.django_db
+def test_send_grant(graphql_client, user, conference_factory, grant_factory):
+    graphql_client.force_login(user)
```
Comment (marcoacierno) [inline backend/api/grants/tests/test_send_grant.py:66]: can you add a test that asserts that the grant request is rejected if the deadline is not set?

File: backend/api/grants/forms.py [hunk 2]
```diff
@@ -1,15 +1,28 @@
...
+    def clean(self):
+        cleaned_data = super().clean()
+
+        conference = cleaned_data.get("conference", None)
+
+        if not conference and self.instance:
+            conference = self.instance.conference
+
+        if not conference.is_grants_closed:
+            raise exceptions.ValidationError(_("The grants form is now closed!"))
```
Comment (marcoacierno) [inline backend/api/grants/forms.py:24]: what happens in the FE when this error appears?

File: backend/conferences/models/conference.py [hunk 3]
```diff
@@ -82,6 +82,15 @@ def is_voting_closed(self):
...
             return False
 
+    @cached_property
+    def is_grants_closed(self):
+        try:
+            grants_deadline = self.deadlines.get(type=Deadline.TYPES.grants)
+
+            return grants_deadline.status == DeadlineStatus.HAPPENING_NOW
+        except Deadline.DoesNotExist:
+            return False
```
[the remark this answers is not in the document]
Comment (estyxx) [reply backend/conferences/models/conference.py:92]: ok but also `is_voting_closed` is wrong to 😱

File: backend/api/grants/forms.py (hunk 2 above)
Comment (estyxx) [reply backend/api/grants/forms.py]: Now we will show 404 page
```javascript
  if (!deadline) {
    return <ErrorPage statusCode={404} />;
  }
```

**Review by marcoacierno — APPROVED** [review]

**Pull request merged**